task06/controller/product.controller.ts

```diff
@@ -0,0 +1,23 @@
+import { Request, Response } from 'express';
+import { fetchProducts, fetchProduct } from '../service/product.service';
+import { AppError } from '../errors/appError';
+import { HttpStatusCode } from '../errors/statusCode';
+
+export const getProducts = (req: Request, res: Response) => {
+  const products = fetchProducts();
+  res.send({ data: products, error: null });
+};
+
+export const getProduct = (req: Request, res: Response) => {
+  const product = fetchProduct(req.params.id);
+
+  if (!product) {
+    throw new AppError(
+      HttpStatusCode.NOT_FOUND,
+      'No product with such id',
+      true
+    );
```

Comment:
FYI: You can also find approach with the `next(error)` function inside controllers, since it will transfer execution to global error handler.